tests/test_sqs/test_sqs.py

```diff
@@ -2742,7 +2747,8 @@ def test_receive_messages_with_message_group_id_on_visibility_timeout():
     with freeze_time("2015-01-01 12:00:00"):
         sqs = boto3.resource("sqs", region_name="us-east-1")
         queue = sqs.create_queue(
-            QueueName="test-queue.fifo", Attributes={"FifoQueue": "true"}
+            QueueName="test-queue.fifo",
+            Attributes={"FifoQueue": "true", "ContentBasedDeduplication": "true"},
```

Comment:
Are there any tests where this is set to `false`?

Reply:
The [default ](https://github.com/getmoto/moto/blob/0520326626ab796e0dd7953a0bf6ac85f96a6d8b/moto/sqs/models.py#L273) for `ContentBasedDeduplication` is `false`, so all the cases where `ContentBasedDeduplication` is not explicitly set `true` for a fifo queue AND a message is sent test for that scenario.

Apart from the tests I added, here is a few:

1. https://github.com/getmoto/moto/blob/0520326626ab796e0dd7953a0bf6ac85f96a6d8b/tests/test_sqs/test_sqs.py#L572
2. https://github.com/getmoto/moto/blob/0520326626ab796e0dd7953a0bf6ac85f96a6d8b/tests/test_sqs/test_sqs.py#L1315
3. https://github.com/getmoto/moto/blob/0520326626ab796e0dd7953a0bf6ac85f96a6d8b/tests/test_sqs/test_sqs.py#L1673
4. https://github.com/getmoto/moto/blob/0520326626ab796e0dd7953a0bf6ac85f96a6d8b/tests/test_sqs/test_sqs.py#L2086
 

I think mine is a breaking change: it will break existing tests where `ContentBasedDeduplication` is not explicitly set to `true` OR  `MessageDeduplicationId` is not passed. However, it reflects more accurately what happens in AWS. 

Ready to merge I reckon 🤘